src/engine/resolvers.rs

```diff
@@ -80,6 +81,65 @@ where
         }
     }
 
+    /// Returns the resolver input deserialized into a structure of type T that
+    /// implements the serde `Deserialize` trait.
+    ///
+    /// # Errors
+    ///
+    /// Returns an [`Error]` variant [`InputNotFound`] if no input field was passed
+    /// to the query, [`TypeConversionFailed`] if unable to convert a warpgrapher Value
+    /// to a serde_json Value, and [`JsonDeserializationFaild`] if unable to parse the
+    /// input data into a struct of type T.
+    ///
+    /// [`Error`]: ../../error/enum.Error.html
+    /// [`InputNotFound`]: ../../error/enum.Error.html#variant.InputNotFound
+    /// [`TypeConversionFailed`]: ../../error/enum.Error.html#variant.TypeConversionFailed
+    /// [`JsonDeserializationFailed`]: ../../error/enum.Error.html#variant.JsonDeserializationFailed
+    pub fn parse_input<T: serde::de::DeserializeOwned>(&self) -> Result<T, Error> {
+        let value: Value = match self.args().get("input") {
```

Comment:
You have a series of steps that each return `Some(v) => v` and/or `Ok(v) => v` and an intermediate value, `value2` that doesn't get used for anything but feeding into the next step.  Admittedly, I haven't written it out, so I may be wrong, but I would guess that with a combination of the `?` operator and some chaining functions like `and_then`, you could probably cut out some complexity.

Reply:
Yes. I originally did it that way, and it was a very ugly looking block of code that made it hard to understand what was happening. Since the rust compiler optimizes the intermediate variable away, I figured this was a more readable approach. But I have no objection to changing it back if you think it would be preferable.

Reply:
I suspect it's just stylistic preference, really, not something that would affect maintainability or, as you noted, the compiled code.  

If you keep the intermediate state and don't chain them together, it might be worth using some amount of `?`, `ok_or_else`, and `map_err` to simplify out some of the match arms that only wrap or unwrap something in an `Ok`, though.  Up to you.

Reply:
Fixed.